libs/core/langchain_core/runnables/base.py

```diff
@@ -1344,6 +1344,36 @@ def with_retry(
     ) -> Runnable[Input, Output]:
         """Create a new Runnable that retries the original runnable on exceptions.
 
+        Example:
+
+        .. code-block:: python
+            from typing import Union
+
+            import pytest
+            from langchain_core.runnables import Runnable, RunnableLambda
+            from pytest_mock import MockerFixture
```

Comment:
Thanks for this!

Ideally this code is easily executed by users getting started with `.with_retry`. Here we require `pytest` and `pytest_mock` for running the example. Could you modify this to be more easily run without requiring our test dependencies? e.g.,
```python
from langchain_core.runnables import RunnableLambda


count = 0

def _lambda(x: int) -> None:
    global count
    count = count + 1
    if x == 1:
        raise ValueError("x is 1")
    else:
        pass

runnable = RunnableLambda(_lambda)
try:
    runnable.with_retry(
        stop_after_attempt=2,
        retry_if_exception_type=(ValueError,),
    ).invoke(1)
except ValueError:
    pass

assert(count == 2)
```